### PR TITLE
[fronius] Fix DecimalType constructor change

### PR DESCRIPTION
**Pull request by jimtng**
Fronius binding is broken due to https://github.com/openhab/openhab-core/pull/2596

### CODE REVIEW

File: bundles/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/internal/handler/FroniusBaseThingHandler.java [hunk 1]
```diff
@@ -107,14 +107,15 @@ protected void updateChannel(String channelId) {
...
+        logger.trace("updateChannel channelId: {} value = {}", channelId, value);
 
         State state = null;
         if (value instanceof BigDecimal) {
             state = new DecimalType((BigDecimal) value);
         } else if (value instanceof Integer) {
             state = new DecimalType(BigDecimal.valueOf(((Integer) value).longValue()));
         } else if (value instanceof Double) {
-            state = new DecimalType((double) value);
+            state = new DecimalType((Double) value);
```
Comment (J-N-K) [inline bundles/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/internal/handler/FroniusBaseThingHandler.java:118]: You should be able to check `instanceof Number` and cast to `(Number)` for all of these cases.

Comment (J-N-K) [reply bundles/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/internal/handler/FroniusBaseThingHandler.java]: `QuantityType` is also a sub-class of `Number`.

Comment (jimtng) [reply bundles/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/internal/handler/FroniusBaseThingHandler.java]: Thanks again. I didn't realise that. I've pushed another one that moved QuantityType before Number.

**Comment by J-N-K** [comment]
I would suggest to wait with merging until it is decided if the core PR is reverted.

**Comment by jimtng** [comment]
Even though it has now been reverted, it appears that going forward it would eventually be removed. Whilst it will compile, shouldn't it just be changed to `Number` in the bindings?

**Review by lolodomo — APPROVED** [review]
LGTM

**Comment by lolodomo** [comment]
@J-N-K : is it ok to merge this PR ?

**Comment by lolodomo** [comment]
@cweitkamp or @wborn : is it ok to merge this PR?

**Comment by cweitkamp** [comment]
No objections from my side. Feel free to merge this PR.

**Pull request merged**